### PR TITLE
fix(examples): Miscellaneous `hf_serialization.py` fixes

### PR DESCRIPTION
**Pull request by Eta0**
# `hf_serialization.py` Fixes

Miscellaneous fixes and improvements for #73.

- Fixes calling `serialize_model` with a default `model_prefix`
- Fixes indentation for saved config files
- Fixes checking local files for emptiness
- Doesn't use `lazy_load=True` when it's not helpful
- Uses a more accurate timing function
- Refactors S3 stream credential propagation

### CODE REVIEW

File: examples/hf_serialization.py [hunk 1]
```diff
@@ -120,19 +121,15 @@ def serialize_model(
...
             logger.info(f"Writing config to {config_path}")
-            with stream_io.open_stream(
-                config_path, "wb+", *s3_write_credentials
-            ) as f:
+            with _write_stream(config_path) as f:
                 config_dict = (
                     config.to_dict() if hasattr(config, "to_dict") else config
                 )
-                f.write(json.dumps(config_dict).encode("utf-8"))
+                f.write(json.dumps(config_dict, indent=2).encode("utf-8"))
```
Comment (sangstar) [inline examples/hf_serialization.py:128]: What's the purpose of this change?

**Review by sangstar — APPROVED** [review]
LGTM

File: examples/hf_serialization.py (hunk 1 above)
Comment (Eta0) [reply examples/hf_serialization.py:129]: It was there before this PR. HuggingFace `config.json` files are typically indented.
